x/warp/contract.go

```diff
@@ -148,60 +167,32 @@ func UnpackGetVerifiedWarpBlockHashOutput(output []byte) (GetVerifiedWarpBlockHa
 	return outputStruct, err
 }
 
-func getVerifiedWarpBlockHash(accessibleState contract.AccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
-	remainingGas, err = contract.DeductGas(suppliedGas, GetVerifiedWarpMessageBaseCost)
-	if err != nil {
-		return nil, remainingGas, err
-	}
-	predicateBytes, valid := unpackWarpMessage(accessibleState, input)
-	// If there is no such value or it failed verification, return invalid
-	if !valid {
-		packedOutput, err := PackGetVerifiedWarpBlockHashOutput(GetVerifiedWarpBlockHashOutput{
-			Valid: false,
-		})
-		if err != nil {
-			return nil, remainingGas, err
-		}
-		return packedOutput, remainingGas, nil
-	}
+type blockHashHandler struct{}
```

Comment:
would it make sense to move those into a message_handler file?

Reply:
Yup tried this out and think it looks a lot better. Great call